ibm_catalog.json

```diff
@@ -93,7 +93,366 @@
+          },
+          "configuration": [
+            {
+              "key": "ibmcloud_api_key",
+              "type": "password"
+            },
+            {
+              "key": "prefix"
+            },
+            {
+              "key": "skip_iam_account_settings"
+            },
+            {
+              "key": "access_token_expiration"
+            },
+            {
+              "key": "active_session_timeout"
+            },
+            {
+              "key": "allowed_ip_addresses"
+            },
+            {
+              "key": "api_creation",
+              "options": [
+                {
+                  "displayname": "Option 1",
+                  "value": "RESTRICTED"
+                },
+                {
+                  "displayname": "Option 2",
+                  "value": "NOT_RESTRICTED"
+                },
+                {
+                  "displayname": "Option 3",
+                  "value": "NOT_SET"
+                }
+              ]
+            },
+            {
+              "key": "enforce_allowed_ip_addresses"
+            },
+            {
+              "key": "inactive_session_timeout"
+            },
+            {
+              "key": "max_sessions_per_identity"
+            },
+            {
+              "key": "mfa",
+              "options": [
+                {
+                  "displayname": "Option 1",
+                  "value": "TOTP4ALL"
+                },
+                {
+                  "displayname": "Option 2",
+                  "value": "TOTP"
+                },
+                {
+                  "displayname": "Option 3",
+                  "value": "LEVEL1"
+                },
+                {
+                  "displayname": "Option 4",
+                  "value": "LEVEL2"
+                },
+                {
+                  "displayname": "Option 5",
+                  "value": "LEVEL3"
+                },
+                {
+                  "displayname": "Option 6",
+                  "value": "NONE"
+                }
+              ]
+            },
+            {
+              "key": "public_access_enabled"
+            },
+            {
+              "key": "refresh_token_expiration"
+            },
+            {
+              "key": "serviceid_creation",
+              "options": [
+                {
+                  "displayname": "Option 1",
+                  "value": "RESTRICTED"
+                },
+                {
+                  "displayname": "Option 2",
+                  "value": "NOT_RESTRICTED"
+                },
+                {
+                  "displayname": "Option 3",
+                  "value": "NOT_SET"
+                }
+              ]
+            },
+            {
+              "key": "shell_settings_enabled"
+            },
+            {
+              "key": "skip_cloud_shell_calls"
+            },
+            {
+              "key": "user_mfa",
+              "type": "array"
```

Comment:
no need to set the type

Reply:
Can't we add a dropdown box here with the supported values?

Reply:
I think this may be one instance where type may need to be specified because the underlying variable is a `set` and whenever I was testing the catalog was defaulting this to a string. I will remove the type to verify but if the catalog does default to string we will need to override here OR we can change the variable from a set to a list and find another way to enforce uniqueness

Reply:
oh - well how about add the custom config with the dropdown for the supported values in the string format thats expected, and don't explicitly set any type. That should work. We can test it out too after you update this branch

Reply:
So this variable is actually expecting a list of objects that looks like

```
[
  {
    iam_id = "IBMid-asdfqwerzxcv"
    mfa = LEVEL3
  },
  {
    iam_id = "IBMid-zxcvqwerasdf"
    mfa = LEVEL1
  }
]
```
setting per user MFA settings so a list (array for json) is the correct type, but whenever the type is not set in the catalog level it defaults to the string type.